### DISCUSSION TITLE
Contiki File System (CFS) on Contiki-NG

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#1964] [Q&A] [answered]
I don't see Contiki File System (CFS) in the documentation [Storage systems](https://github.com/contiki-ng/contiki-ng/wiki#storage-systems) anymore, only Coffee. Is CFS supported in NG? 

Also, is there a drawback for using Coffee or Antelope while running in .native mode?

**Comment by nvt** [comment, marked as answer]
The Coffee documentation covers the basic CFS interface, as well as Coffee extensions to the CFS interface. The only other option to using Coffee as an implementation for the CFS interface is to use cfs-posix on the native platform. The CFS-Posix implementation is just a wrapper for the corresponding POSIX file and directory functions.

A key difference of the two implementations for the native platforms is that the Coffee implementation will write a Coffee file structure in RAM, whereas the cfs-posix implementation will create individual files in the host file system under a certain directory.

It is possible to switch from the default cfs-posix to Coffee on the native platform by adding the statement

<code>MAKE_CFS = MAKE_CFS_COFFEE</code>

to your project's makefile.

**Reply by rahav** [reply]
This is very helpful. Thank you.
